### PR TITLE
Migrate most of search component to ts 

### PR DESCRIPTION
**Pull request by haoyxc**
everything except SearchBar.tsx

also changed the filters to booleans instead of 0/1s

### CODE REVIEW

File: frontend/plan/actions/index.js [hunk 1]
```diff
@@ -211,7 +211,7 @@ function buildCourseSearchUrl(filterData) {
     const reqs = [];
     if (filterData.selectedReq) {
         for (const key of Object.keys(filterData.selectedReq)) {
-            if (filterData.selectedReq[key] === 1) {
+            if (filterData.selectedReq[key]) { // change from == 1
```
Comment (davish) [inline frontend/plan/actions/index.js:214]: No need to add comments with what changed :)
```suggestion
            if (filterData.selectedReq[key]) {
```

File: frontend/plan/actions/index.js [hunk 2]
```diff
@@ -267,7 +267,7 @@ function buildCourseSearchUrl(filterData) {
             const applied = [];
             Object.keys(filterData[checkboxFields[i]]).forEach((item) => {
                 // eslint-disable-line
-                if (filterData[checkboxFields[i]][item] === 1) {
+                if (filterData[checkboxFields[i]][item]) { // changed from ===1
```
Comment (davish) [inline frontend/plan/actions/index.js:270]: ditto here

File: frontend/plan/components/search/CheckboxFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                                </label>
+                            </div>
+                        </div>
+                    );
+                })}
+        </div>
+    );
+}
+
+CheckboxFilter.propTypes = {
```
Comment (davish) [inline frontend/plan/components/search/CheckboxFilter.tsx:88]: remember to remove proptypes after converting files

File: frontend/plan/components/search/SchoolReq.tsx [hunk 4]
```diff
@@ -1,24 +1,43 @@
...
+
+interface SchoolReqProps {
+    startSearch: (searchObj: object) => void;
+    filterData: FilterData;
+    schoolReq: { [K in School]: Requirement[] };
+    addSchoolReq: (s: string) => void;
+    remSchoolReq: (s: string) => void;
+}
+
+type SchoolDisplay = "College" | "Engineering" | "Nursing" | "Wharton";
```
Comment (davish) [inline frontend/plan/components/search/SchoolReq.tsx:13]: Is there an enum that could be used for this type? would help with the repitition between here and the actual list.

File: frontend/plan/components/search/SchoolReq.tsx [hunk 5]
```diff
@@ -1,24 +1,43 @@
...
+    ];
+    const [selSchool, setSelSchool] = useState<SchoolDisplay>("College");
 
-    const schoolCode = new Map();
-    schoolCode.set("College", "SAS");
-    schoolCode.set("Engineering", "SEAS");
-    schoolCode.set("Wharton", "WH");
-    schoolCode.set("Nursing", "NURS");
+    const schoolCode: { [K in SchoolDisplay]: School } = {
+        College: School.COLLEGE,
```
Comment (davish) [inline frontend/plan/components/search/SchoolReq.tsx:31]: @eyingxuan can correct me if I'm wrong, but I believe that if you give enums a string value, you can just cast the string to the enum type and things will work as expected.

File: frontend/plan/components/search/SearchBar.tsx [hunk 6]
```diff
@@ -0,0 +1,450 @@
+// import React, { useEffect, useState } from "react";
```
Comment (davish) [inline frontend/plan/components/search/SearchBar.tsx:1]: i would delete this file in this PR if it's 100% commented out and you've got another PR open

**Review by davish — APPROVED** [review]
Looks great! Some small changes, but after that go for the merge!!!

File: frontend/plan/components/search/SchoolReq.tsx (hunk 5 above)
Comment (eyingxuan) [reply frontend/plan/components/search/SchoolReq.tsx:31]: I think the issue here was that the backend requires using an abbreviation of each school in the route (such as "SAS", "SEAS" etc) so School.COLLEGE is set to be equal to the abbreviated string, while we need what's displayed to be an actual string. 

What could be cleaner in retrospect is actually to write a function that turns the School enum into a display string, and use the enum itself in the useState to change between school requirements. What do you think? @davish @haoyxc

Comment (haoyxc) [reply frontend/plan/components/search/SchoolReq.tsx:31]: i'll defer to you guys since you probably know the codebase better than I do :-)

**Pull request merged**